pac4j-core/src/main/java/org/pac4j/core/profile/service/InMemoryProfileService.java

```diff
@@ -70,18 +71,18 @@ protected void deleteById(final String id) {
 	@Override
 	protected List<Map<String, Object>> read(final List<String> names, final String key, final String value) {
 		logger.debug("Reading key / value: {} / {}", key, value);
-		final List<Map<String, Object>> listAttributes = new ArrayList<>();
+		final List<Map<String, Object>> listAttributes;
 		if (key.equals(getIdAttribute())) {
+			listAttributes = new ArrayList<>();
 			final Map<String,Object> profile = profiles.get(value);
```

Comment:
It's just a personal preference, mostly for readability yes, but it's ok if you prefer not to use it.

Regarding this particular use case, you are completely right, I read too fast and misunderstood what was written :)